### PR TITLE
Modify async target re-reg and GET test

### PR DESCRIPTION
**Pull request by liangdrew**
Now, split the GET errors into two categories:

1) GET errors occurring before the target has updated its local bucket map
2) GET errors occurring after the target has updated its local bucket map

### CODE REVIEW

File: dfc/tests/regression_test.go [hunk 1]
```diff
@@ -260,38 +263,56 @@ func TestGetAndReRegisterInParallel(t *testing.T) {
 				}
 				_, _, err := client.Get(proxyurl, bucket, SmokeStr+"/"+repFile.filename, nil, nil, false, false)
 				if err != nil {
-					atomic.AddUint64(&numGetErrs, 1)
+					if reregistered == 1 {
```
Comment (VladimirMarkelov) [inline dfc/tests/regression_test.go]: reregistered is set using atomics but read without it

File: dfc/tests/regression_test.go [hunk 2]
```diff
@@ -260,38 +263,58 @@ func TestGetAndReRegisterInParallel(t *testing.T) {
...
+					if r == 1 {
+						atomic.AddUint64(&numGetErrsAfter, 1)
+					} else {
+						atomic.AddUint64(&numGetErrsBefore, 1)
+					}
 				}
 				atomic.AddUint64(&getsCompleted, 1)
-				if (getsCompleted)%5000 == 0 {
-					tlogf(" %d/%d GET requests completed...\n", getsCompleted, num*numGetsForEachFile)
+				g := atomic.LoadUint64(&getsCompleted)
```
Comment (VladimirMarkelov) [inline dfc/tests/regression_test.go]: nitpick: AddUint64 returns a new value, so you can combine two lines in one

Comment (liangdrew) [reply dfc/tests/regression_test.go]: I used a variable here because the value is used twice below

Comment (VladimirMarkelov) [reply dfc/tests/regression_test.go]: I meant this line and above, not below )

**Pull request merged**